### PR TITLE
feat: Add a debug feature to dump content book data

### PR DESCRIPTION
**Pull request by kristofbolyai**

### CODE REVIEW

File: common/src/main/java/com/wynntils/features/debug/ContentBookDumpFeature.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        JsonElement element = Managers.Json.GSON.toJsonTree(mappedActivities);
+
+        String fileName = "content_book_dump.json";
+        File jsonFile = new File(SAVE_FOLDER, fileName);
+        Managers.Json.savePreciousJson(jsonFile, element.getAsJsonObject());
+
+        McUtils.sendMessageToClient(Component.literal("Saved content book dump to " + jsonFile.getAbsolutePath()));
+    }
+
+    // This method creates a consisnent mapping and ordering for the dumped activities
```
Comment (magicus) [inline common/src/main/java/com/wynntils/features/debug/ContentBookDumpFeature.java]:
```suggestion
    // This method creates a consistent mapping and ordering for the dumped activities
```

File: common/src/main/java/com/wynntils/features/debug/ContentBookDumpFeature.java [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        Map<ActivityType, List<DumpableActivityInfo>> mappedDumpedActivities = new EnumMap<>(ActivityType.class);
+        for (ActivityType value : ActivityType.values()) {
+            mappedDumpedActivities.put(value, new ArrayList<>());
+        }
+
+        for (DumpableActivityInfo activityInfo : currentDump) {
+            ActivityType type = activityInfo.type();
+
+            // Merge STORYLINE_QUEST and QUEST
+            if (type == ActivityType.STORYLINE_QUEST) {
```
Comment (magicus) [inline common/src/main/java/com/wynntils/features/debug/ContentBookDumpFeature.java:196]: I think we can keep this. I did not notice when looking at the json dump that it was missing. This is a distinction Wynn makes, there is no reason to throw away information at this stage. We can merge this later on, if we want.

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/features/debug/ContentBookDumpFeature.java:196]: We do not throw it away. The type is still there, we just group it in the dump.

**Comment by kristofbolyai** [comment]
https://github.com/Wynntils/Data-Storage/pull/36

File: common/src/main/java/com/wynntils/features/debug/ContentBookDumpFeature.java (hunk 2 above)
Comment (magicus) [reply common/src/main/java/com/wynntils/features/debug/ContentBookDumpFeature.java:196]: Aha, I see.

**Review by magicus — APPROVED** [review]

**Pull request merged**